### PR TITLE
Provide a way to un-sync (#64)

### PR DESCRIPTION
**Pull request by nkoterba**
Currently, vuex-router-sync only publicly exposes a single method: ‘sync’.  This method creates a store.watch as well as a router.beforeHook.  No publicly available method existed to effectively ‘unsync’ vuex-router-sync and remove the watch and hook, which created ‘dangling’ refrences and a potential memory leak.  Users of this library may need an ‘unsync’ function if they, for example, only use Vue.js in specific portions of their webapp (e.g. with React, Angular, etc.) and when users navigate away from that portion they want to remove/clean-up all Vue.js resources and components.  Includes a unit test.  Had to bump vue-router to version 2.5.0 which was the first version to allow de-registering hook functions.

### CODE REVIEW

File: package.json
Comment (posva) [inline package.json]: don't change the version, we will change this when releasing 🙂

Comment (nkoterba) [reply package.json]: Ok. I've reset my head prior to the version change.

**Review by LinusBorg — APPROVED** [review]
LGTM

**Comment by LinusBorg** [comment]
@posva: We can merge, agreed?

**Comment by posva** [comment]
@LinusBorg 👍 ok for me! Shouldn't we wait for a little review from Evan?

**Comment by yyx990803** [comment]
Thanks!

**Pull request merged**
